### PR TITLE
Simplify some conditions

### PR DESCRIPTION
**Pull request by monosans**

### CODE REVIEW

**Comment by az0** [comment]
Thank you, monosans. 

There are many changes here, so this will need to wait for evaluation, testing, and to avoid conflict with a large change that Max is working on to update Python on Windows.

**Comment by az0** [comment]
For cross reference, I mean #848 . A few days ago, Max emailed me that he has made big progress

File: bleachbit/FileUtilities.py [hunk 1]
```diff
@@ -645,18 +645,17 @@ def is_dir_empty(dirname):
...
-            for entry in it:
+    if hasattr(os, 'scandir'):
+        if sys.version_info < (3, 6, 0):
+                    # Python 3.5 added os.scandir() without context manager.
+            for _ in os.scandir(dirname):
                 return False
+        else:
+            # Python 3.6 added the context manager.
+            with os.scandir(dirname) as it:
+                for entry in it:
```
Comment (az0) [inline bleachbit/FileUtilities.py:656]: `entry` could be `_`

File: bleachbit/markovify/utils.py [hunk 2]
```diff
@@ -50,14 +50,13 @@ def combine(models, weights=None):
     if isinstance(ret_inst, Chain):
         return Chain.from_json(c)
     if isinstance(ret_inst, Text):
-        if any(m.retain_original for m in models):
```
Comment (az0) [inline bleachbit/markovify/utils.py:53]: You may want to suggest the same to the upstream project https://github.com/jsvine/markovify

**Review by az0 — APPROVED** [review]
I wrote one small suggestion in case you want to follow up later

**Comment by az0** [comment]
@monosans 
Thank you for the changes. I am just curious: do you use a code profiling tool to find these, or do you just read carefully?

File: bleachbit/FileUtilities.py (hunk 1 above)
Comment (monosans) [reply bleachbit/FileUtilities.py:656]: > `entry` could be `_`

Challenge accepted :) #1228

**Comment by monosans** [comment]
> do you use a code profiling tool to find these, or do you just read carefully?

For many of these refactors I use [wemake-python-styleguide](https://github.com/wemake-services/wemake-python-styleguide). This linter is incredibly good.

**Pull request merged**